### PR TITLE
solve both puzzles on day 10

### PR DESCRIPTION
**Pull request by jmcda001**

### CODE REVIEW

File: days/src/Main.kt [hunk 1]
```diff
@@ -10,14 +11,25 @@ import day9.Day9
...
-    println("Day 1: ${evaluateDay(Day1.buildFromFile("day1/input.txt"))}")
-    println("Day 2: ${evaluateDay(Day2.buildFromFile("day2/input.txt"))}")
-    println("Day 3: ${evaluateDay(Day3.buildFromFile("day3/input.txt"))}")
-    println("Day 4: ${evaluateDay(Day4.buildFromFile("day4/input.txt"))}")
-    println("Day 5: ${evaluateDay(Day5.buildFromFile("day5/input.txt"))}")
-    println("Day 6: ${evaluateDay(Day6.buildFromFile("day6/input.txt"))}")
-    println("Day 7: ${evaluateDay(Day7.buildFromFile("day7/input.txt"))}")
-    println("Day 8: ${evaluateDay(Day8.buildFromFile("day8/input.txt"))}")
-    println("Day 9: ${evaluateDay(Day9.buildFromFile("day9/input.txt"))}")
+    val runAll = false
```
Comment (brrcrites) [inline days/src/Main.kt:27]: You could use an execution parameter here and save yourself the re-compilation (its really hard to find docs on this for some reason)

https://www.topjavatutorial.com/kotlin/reading-command-line-arguments-in-kotlin/

File: days/src/day10/Day10.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package day10
+
+import Day
+import java.io.File
+
+class Day10(values: List<Int>) : Day {
+    private val jolts = values.sorted().let { listOf(0) + it + it.last() }
```
Comment (brrcrites) [inline days/src/day10/Day10.kt:7]: Sorting first was smart, I just build a range and scanned for candidates each time

File: days/src/day10/Day10.kt [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    override fun puzzle1(): Int? = jolts.mapIndexed { i, jolt ->
+            jolt - if (i > 0) { jolts[i-1] } else { 0 }
+        }.filterNot { it == 2 }.partition { it == 1 }.let { (ones, threes) ->
+            ones.size * (threes.size)
+        }
```
Comment (brrcrites) [inline days/src/day10/Day10.kt:12]: This partition is sweet too, wish we had this in JS

**Review by brrcrites — APPROVED** [review]

File: days/src/Main.kt (hunk 1 above)
Comment (jmcda001) [reply days/src/Main.kt:27]: Added on next PR

**Pull request merged**